### PR TITLE
✨ Add 'EnterToSend' Option & Update Brazilian Portuguese Translation 🇧🇷

### PR DESCRIPTION
**Pull request by itzraiss**
## Summary

This PR introduces the 'EnterToSend' feature, allowing users to choose whether to send messages with the Enter key, for those using LibreChat on mobile, it enables line breaks. Additionally, it includes updates to the Brazilian Portuguese translation.

### Features:
- **EnterToSend Toggle**: Users can now decide their preferred method of sending messages, either by clicking the send button or using the Enter key, improving the app's usability.

_Note: it is already active by default_

### Translation Portuguese Brazilian
- **Added new translations**

**Images:**
image

**Demo Video:**

https://github.com/danny-avila/LibreChat/assets/140329135/0a4b15d2-3f16-469b-be87-6d800a1bd024

## Change Type

- [x] New feature (non-breaking change which adds functionality)
- [x] Translation update

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes

### CODE REVIEW

File: client/src/hooks/Input/useTextarea.ts [hunk 1]
```diff
@@ -109,16 +106,9 @@ export default function useTextarea({
     }
 
     const getPlaceholderText = () => {
-      if (
```
Comment (danny-avila) [inline client/src/hooks/Input/useTextarea.ts]: Why does this get removed?

File: client/src/hooks/Input/useTextarea.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import debounce from 'lodash/debounce';
 import React, { useEffect, useRef, useCallback } from 'react';
-import { EModelEndpoint } from 'librechat-data-provider';
-import type { TEndpointOption } from 'librechat-data-provider';
+import { useRecoilState } from 'recoil';
+import store from '~/store';
```
Comment (danny-avila) [inline client/src/hooks/Input/useTextarea.ts:8]: Please bring back what you removed and organize imports by length:

First, library/package imports, from shortest to longest,
Then, types from longest to shortest
Last, local project files from longest to shortest

File: client/src/hooks/Input/useTextarea.ts [hunk 3]
```diff
@@ -158,26 +148,27 @@ export default function useTextarea({
...
-      e.preventDefault();
-    }
-
-    if (isNonShiftEnter && !isComposing?.current) {
-      submitButtonRef.current?.click();
+    if (e.key === 'Enter' && !enterToSend) {
+      if (!e.shiftKey) {
+        e.preventDefault();
+        if (textAreaRef.current && textAreaRef.current.value !== null) {
+          insertTextAtCursor(textAreaRef.current, '\n');
```
Comment (danny-avila) [inline client/src/hooks/Input/useTextarea.ts]: not sure why this is necessary, seems to work without `insertTextAtCursor`?

File: client/src/localization/languages/Eng.tsx [hunk 4]
```diff
@@ -322,6 +322,7 @@ export default {
   com_nav_welcome_assistant: 'Please Select an Assistant',
   com_nav_welcome_message: 'How can I help you today?',
   com_nav_auto_scroll: 'Auto-scroll to Newest on Open',
+  com_nav_enter_to_send: 'Do Send Message with Enter Key?',
```
Comment (danny-avila) [inline client/src/localization/languages/Eng.tsx]: change to "Send messages with Enter Key"

File: client/src/hooks/Input/useTextarea.ts [hunk 5]
```diff
@@ -158,26 +148,27 @@ export default function useTextarea({
...
-    }
-
-    if (isNonShiftEnter) {
-      e.preventDefault();
-    }
-
-    if (isNonShiftEnter && !isComposing?.current) {
-      submitButtonRef.current?.click();
+    if (e.key === 'Enter' && !enterToSend) {
+      if (!e.shiftKey) {
```
Comment (danny-avila) [inline client/src/hooks/Input/useTextarea.ts]: Please try to avoid nesting if statements

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by danny-avila** [comment]
Thanks for the PR! Finally got around to reviewing and there are some changes requested

File: client/src/hooks/Input/useTextarea.ts (hunk 2 above)
Comment (itzraiss) [reply client/src/hooks/Input/useTextarea.ts:8]: solved in 2832434ba157810b771043d729b5ff60346ce278

File: client/src/hooks/Input/useTextarea.ts (hunk 3 above)
Comment (itzraiss) [reply client/src/hooks/Input/useTextarea.ts]: solved  2832434ba157810b771043d729b5ff60346ce278

File: client/src/localization/languages/Eng.tsx (hunk 4 above)
Comment (itzraiss) [reply client/src/localization/languages/Eng.tsx]: updated 2832434ba157810b771043d729b5ff60346ce278

File: client/src/hooks/Input/useTextarea.ts [hunk 6]
```diff
@@ -185,22 +189,23 @@ export default function useTextarea({
...
-    if (isNonShiftEnter && filesLoading) {
-      e.preventDefault();
-    }
+    const isEnterKey = e.key === 'Enter';
+    const isNonShiftEnter = isEnterKey && !e.shiftKey;
 
-    if (isNonShiftEnter) {
+    if (isEnterKey && (isSubmitting || filesLoading)) {
       e.preventDefault();
+      return;
```
Comment (danny-avila) [inline client/src/hooks/Input/useTextarea.ts]: No early `return;` is necessary here

Comment (itzraiss) [reply client/src/hooks/Input/useTextarea.ts]: fix: a36b7fa7e1f798e57a649b1cc70f1444326fbd9d

**Comment by danny-avila** [comment]
Closing in favor of #2413

**Pull request closed without merging**